gradle/plugins/src/main/kotlin/net/twisterrob/sun/plugins/MergeLintSarifReportsTask.kt

```diff
@@ -0,0 +1,165 @@
+package net.twisterrob.sun.plugins
+
+import io.github.detekt.sarif4k.ArtifactLocation
+import io.github.detekt.sarif4k.ReportingDescriptor
+import io.github.detekt.sarif4k.Result
+import io.github.detekt.sarif4k.Run
+import io.github.detekt.sarif4k.SarifSchema210
+import io.github.detekt.sarif4k.SarifSerializer
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+
+/**
+ * Based on Detekt's SARIF merging task.
+ *
+ * @see io.gitlab.arturbosch.detekt.report.ReportMergeTask
+ * @see io.gitlab.arturbosch.detekt.report.SarifReportMerger
+ */
+@CacheableTask
+abstract class MergeLintSarifReportsTask : DefaultTask() {
+
+	@get:InputFiles
+	@get:PathSensitive(PathSensitivity.RELATIVE)
+	abstract val sarifFiles: ConfigurableFileCollection
+
+	@get:OutputFile
+	abstract val mergedSarifFile: RegularFileProperty
+
+	@TaskAction
+	fun merge() {
+		val inputs = sarifFiles.files
+		logger.info("Inputs")
+		logger.info(inputs.joinToString(separator = "\n") { "${it.absolutePath} (exists=${it.exists()})" })
+
+		val output = mergedSarifFile.get().asFile
+		logger.info("Output = ${output.absolutePath}")
+
+		merge(inputs.filter { it.exists() }, output)
+		logger.lifecycle("Merged SARIF output to ${output.absolutePath}")
+	}
+
+	private fun merge(inputs: Collection<File>, output: File) {
+		val sarifs = inputs.associateWith { SarifSerializer.fromJson(it.readText()) }
+
+		val runs = sarifs.mapValues { it.value.runs.size }
+		require(runs.values.all { it == 1 }) {
+			"Some sarifs have no or multiple runs:\n${runs.filterValues { it != 1 }}"
+		}
+
+		val tools = sarifs.mapValues { it.value.run.tool.driver.name }
+			.entries.groupBy({ it.value }, { it.key })
+		require(tools.size == 1) {
+			"Cannot merge sarifs from different tools:\n$tools"
+		}
+
+		val commonPath = commonSarifPath(sarifs.values)
```

Comment:
what if this is empty or equal to path?

Reply:
empty is meaningless, equal means nothing will change...

which means single failure cannot be located correctly, how to detect "common path" for a single failure? which needs to point to project root.

Reply:
oh, wait, this is Gradle task, so we know absolute path of rootProject 🤦‍♂️

Reply:
https://github.com/TWiStErRob/net.twisterrob.sun/pull/83/commits/9a9afa40d7e009faddadf990195ec6f4220c0453